cmake/CMakeLists.txt

```diff
@@ -141,11 +141,10 @@ option(onnxruntime_FUZZ_TEST "Enable Fuzz testing" OFF)
 # training options
 option(onnxruntime_ENABLE_NVTX_PROFILE "Enable NVTX profile." OFF)
 option(onnxruntime_ENABLE_MEMORY_PROFILE "Enable memory profile." OFF)
-option(onnxruntime_ENABLE_TRAINING "Enable training functionality." OFF)
-# For on-device training ort build, onnxruntime_ENABLE_TRAINING must be ON.
-option(onnxruntime_ENABLE_TRAINING_ON_DEVICE "Enable training on device." OFF)
+option(onnxruntime_ENABLE_TRAINING "Enable full training functionality. Includes ORTModule and ORT Training APIs" OFF)
+# For building with onnxruntime_ENABLE_TRAINING_APIS, onnxruntime_ENABLE_TRAINING must be ON.
+option(onnxruntime_ENABLE_TRAINING_APIS "Enable ort training apis." OFF)
```

Comment:
This comment is a bit confusing as later on `onnxruntime_ENABLE_TRAINING` turns on `onnxruntime_ENABLE_TRAINING_APIS`, and the overall idea of this PR seems to involve making `onnxruntime_ENABLE_TRAINING` a higher level flag that enables lots of functionality. Is the support for just enabling `onnxruntime_ENABLE_TRAINING_APIS` coming in a separate PR? I assume the latter is the option I should be looking for in Java as we can't interop with the `ORTModule` bits in pytorch.

Reply:
Yes, the support for just enabling onnxruntime_ENABLE_TRAINING_APIS is coming in a separate PR.